### ISSUE TITLE
make old dune versions unavailable on win32

### ISSUE THREAD

**Issue by raphael-proust** [ocaml/opam-repository#25483]
> Note for opam-repository maintainers: this release fixes a critical issue for dune on Windows (support of unicode characters in PATH, e.g. usernames with accents) and i would suggest all the older versions of dune to be restricted to `available: os != "win32"` in a separate PR, independently of this one (this one should be merged first though).

_Originally posted by @kit-ty-kate in https://github.com/ocaml/opam-repository/issues/25474#issuecomment-1991984048_

**Comment by kit-ty-kate** [comment]
Reverted in https://github.com/ocaml/opam-repository/pull/25609
